### PR TITLE
add unit tests for policy validations

### LINKED ISSUE #3704: Show Application Violations in WeaveGitOps OSS
As a user using Weave Policy Agent, I need to see the policy violations in my application. 

#### Application Criteria 
* We need to replicate the violations tab in the application view in the Enterprise product.

### PR DESCRIPTION
**Pull request by waleedhammam**
Closes #3704

**What changed?**

- add unit tests for policy validations

**How did you validate the change?**

- Manual testing 
- Unit tests
- CI

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Review by serboctor — APPROVED** [review]

**Comment by serboctor** [comment]
I am not a fan of using things like `ns1`, `ns2`, and `o` as variable names but it's not a dealbreaker as well.

File: core/server/policy_violations_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	cfg := makeServerConfig(client, t, "")
+	c := makeServer(cfg, t)
+	res, err := c.ListPolicyValidations(ctx, &pb.ListPolicyValidationsRequest{
+		Application: "app1",
+		Kind:        "HelmRelease",
+		Namespace:   "weave-system",
+		ClusterName: "Default",
+	})
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(err).To(BeNil())
```
Comment (Samra10) [inline core/server/policy_violations_test.go:131]: It would be better to add a check for res that it's not null.

Comment (waleedhammam) [reply core/server/policy_violations_test.go:131]: checking for null pointer will raise an error

File: core/server/policy_violations_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+			event := o.(*corev1.Event)
+			return []string{event.Type}
+		})).
+		Build()
+
+	cfg := makeServerConfig(client, t, "")
+	c := makeServer(cfg, t)
+	res, err := c.ListPolicyValidations(ctx, &pb.ListPolicyValidationsRequest{})
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(err).To(BeNil())
```
[the remark this answers is not in the document]
Comment (waleedhammam) [reply core/server/policy_violations_test.go:186]: checking for result length here https://github.com/weaveworks/weave-gitops//blob/39b4b430fbe0e8ec49678b7567fba39a1445861e/core/server/policy_violations_test.go#L187

**Pull request merged**